### PR TITLE
Set password for root account to allow remote access.

### LINKED ISSUE #15: Security checkup
Make sure exactly which users get created during initialization, what their access rights are, and so on. This should be mainly done by looking into mysql.user table. We should also document as much as possible the behaviour of (not) setting `MYSQL_ROOT_USER`.

### PR DESCRIPTION
**Pull request by mnagy**
This partially resolves #15. What remains is update the openshift-docs.

Test work fine with both rhel and centos.

### CODE REVIEW

**Comment by bparees** [comment]
lgtm, please make sure the openshift-docs get this update too.

File: 5.5/test/run [hunk 1]
```diff
@@ -92,6 +104,17 @@ run_tests() {
...
 	test_connection $name
+	echo "  Testing login accesses"
+	assert_login_access $USER $PASS true
+	assert_login_access $USER "${PASS}_foo" false
+	if [ -v ROOT_PASS ]; then
+		assert_login_access root $ROOT_PASS true
+		assert_login_access root "${ROOT_PASS}_foo" false
+	else
+		assert_login_access root "foo" false
+		assert_login_access root "" false
```
Comment (bparees) [inline 5.5/test/run]: hmm..  in the docs you imply that this should work: The root user has no password set by default, only allowing local
connections.

File: README.md [hunk 2]
```diff
@@ -64,9 +64,11 @@ present, `mysqld` is executed and will run as PID 1. You can stop the
 detached container by running `docker stop <CONTAINER ID>`.
 
 ### MySQL root user
-The root user has no password set by default. You can set it by setting
-`MYSQL_ROOT_PASSWORD` environment variable when initializing your
-database.
+The root user has no password set by default, only allowing local
+connections. You can set it by setting `MYSQL_ROOT_PASSWORD` environment
```
Comment (jhadvig) [inline README.md]: Shouldn't it be the MYSQL_ADMIN_PASSWORD ?

File: 5.5/run-mysqld.sh [hunk 3]
```diff
@@ -85,10 +85,11 @@ if [ ! -d '/var/lib/mysql/mysql' ]; then
 	EOSQL
 
 	if [ -v root_pass ]; then
-		mysqladmin --socket=/tmp/mysql.sock password "${root_pass}" flush-privileges shutdown
-	else
-		mysqladmin --socket=/tmp/mysql.sock shutdown
+		mysql --socket=/tmp/mysql.sock <<-EOSQL
+			GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' IDENTIFIED BY '${root_pass}';
```
Comment (mfojtik) [inline 5.5/run-mysqld.sh]: Maybe do `FLUSH PRIVILEGES` here to assure the privileges are properly saved into data files?

File: README.md (hunk 2 above)
Comment (mfojtik) [inline README.md]: +1 keep things consistent with Mongo and Postresql

**Comment by mfojtik** [comment]
LGTM, two small nits

File: README.md (hunk 2 above)
Comment (bparees) [inline README.md]: +1

Comment (mnagy) [inline README.md]: We've had a discussion about MySQL using the root user account, which is created by default and agreed by SCL people. Admin account wouldn't be the same as root, where root is more popular and also used by the official MySQL docs.

I don't see how consistency with mongo is relevant. Those are different worlds, most likely different users. And MYSQL_ROOT_PASSWORD is optional, mind you. If someone needs it, they should know what they are doing and they should read the docs where they'll discover that it's ROOT, not ADMIN.

File: 5.5/test/run (hunk 1 above)
Comment (mnagy) [inline 5.5/test/run]: Those are not local connections, the mysql commands are executed from the host. By docker, using the same image, but from the host.

File: 5.5/run-mysqld.sh (hunk 3 above)
Comment (mnagy) [inline 5.5/run-mysqld.sh]: Will fix.

File: 5.5/test/run (hunk 1 above)
Comment (bparees) [inline 5.5/test/run]: got it.

File: README.md (hunk 2 above)
Comment (bparees) [inline README.md]: /me is convinced.

**Comment by mnagy** [comment]
Small change, added flush-privileges before shutdown.

File: README.md (hunk 2 above)
Comment (jhadvig) [inline README.md]: Like to me it doesn't matter what would be the name, ROOT or ADMIN. The point is to stay consistent cause both of those roles, in mysql and mongo are doing the same. Why there should be a different names ?

Comment (mfojtik) [inline README.md]: let's issue this concern and return to it later ;-)

**Pull request merged**